### PR TITLE
Add ignition support

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/roadmap/issues/778

Tested scenarios:

- Creating a fresh cluster with Ubuntu
- Creating a fresh cluster with Flatcar
- Converting Ubuntu cluster to a Flatcar cluster
- Upgrading 0.13.0 Ubuntu cluster to this branch

### CODE REVIEW

**Review by bavarianbidi — APPROVED** [review]

**Comment by bavarianbidi** [comment]
out of curiosity: which container-runtime is flatcar using per default? Do we have to explicity set the path to the socket?

xref: https://flatcar-linux.org/docs/latest/container-runtimes/switching-from-docker-to-containerd-for-kubernetes/

**Comment by erkanerol** [comment]
Good work 👍

File: helm/cluster-openstack/files/etc/gs-kube-proxy-patch.sh [hunk 1]
```diff
@@ -21,5 +22,11 @@ if [[ ! -f ${dir}/gs-kube-proxy-config.yaml ]]; then
...
 
-cat ${dir}/gs-kube-proxy-config.yaml >> ${dir}/kubeadm.yaml
-rm ${dir}/gs-kube-proxy-config.yaml
+# kubeadm config is in different directory in Flatcar (/etc) and Ubuntu (/run/kubeadm).
+kubeadm_file="/etc/kubeadm.yml"
+if [[ ! -f ${kubeadm_file} ]]; then
+    kubeadm_file="/run/kubeadm/kubeadm.yaml"
+fi
+
+cat "${dir}/gs-kube-proxy-config.yaml" >> "${dir}/kubeadm.yaml"
```
Comment (fiunchinho) [inline helm/cluster-openstack/files/etc/gs-kube-proxy-patch.sh:31]: hey @kopiczko , shouldn't this be
```
cat "${dir}/gs-kube-proxy-config.yaml" >> "${kubeadm_file}"
```
?

File: helm/cluster-openstack/files/etc/gs-kube-proxy-patch.sh [hunk 2]
```diff
@@ -9,7 +9,8 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-readonly dir="/run/kubeadm"
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+readonly dir
 
 # Run this script only if this is the init node.
 if [[ ! -f ${dir}/kubeadm.yaml ]]; then
```
Comment (fiunchinho) [inline helm/cluster-openstack/files/etc/gs-kube-proxy-patch.sh:16]: I understood this file was present only on `flatcar`. Would this work on `Ubuntu`? If the file is not present on `Ubuntu` the script will do nothing.

**Pull request merged**